### ISSUE TITLE
editor agnostic package and migration to REditorSupport

### ISSUE THREAD

**Issue by randy3k** [ManuelHentschel/vscDebugger#33]
@ManuelHentschel Thank you for working on this.

I have two comments/questions for you.

1. I am the author of the [languageserver](https://github.com/REditorSupport/languageserver) and creator of the organization REditorSupport. The goal of the organization is to gather all editor related tools. I am wondering if you will be interested in migrating your packages to REditorSupport. (Of course, you will become the admin of your packages) Please let me know if you are interested. I am sure users will be benefited by having all related packages in the same place.

2. I took a detailed look at this package. It does seem that you are not communicating with the editor using the debugger adapter protocol here. The adapter is actually one part of `VSCode-R-Debugger` and it is just a helper package to it.
I reckon that it would be useful to extract the adapter code out of `VSCode-R-Debugger` so other editors can benefit from it
https://github.com/emacs-lsp/dap-mode
https://github.com/dradtke/vim-dap
https://github.com/daveleroy/sublime_debugger
Also see how python adapter is written: https://github.com/microsoft/debugpy

**Comment by randy3k** [comment]
I took a detailed look of the package. It doesn't seem that you are using the debug adapter protocol?

**Comment by ManuelHentschel** [comment]
Thanks for your interest in these projects.

1. Sounds like a good idea to have all the R tools in one place, so I wouldn't migrating the packages there. Plus we could really use the support and knowhow of the other REditorSupport members :)

2. That's mostly because the project started out as a VSCode extension with just a few supporting R functions. Over time I added so much R code that it felt justified to move it into a separate package, but I didn't plan the entire thing as a straight DAP implementation in R. In recent changes I have tried to stick as close as possible to the DAP in the R code so that the extension mostly just passes through the objects received from R.
Once everything works a bit more stable I agree that it would be useful to extract the pure adapter code. Implementing the entire thing in pure R looks rather difficult, though, since e.g. the flow control in the `browser()` is not accessible by normal R functions (as far as I'm aware)
